### PR TITLE
rain: init at 1.2.0

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

https://github.com/aws-cloudformation/rain/releases/tag/v1.2.0

It cannot build with Go 1.18, so Go 1.17 is used.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rain/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "aws-cloudformation";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-6YKZy6sdy1Yi2cDaLMA54GBTZ9uPhYi5Cq5QqCGbD5k=";
+  };
+
+  vendorSha256 = "sha256-e3R8+xarofbx3Ky6JIfDbysTQETCUaQj/QmzAiU7fZk=";
+
+  subPackages = [ "./cmd/rain" ];
```
Comment (azahi) [inline pkgs/development/tools/rain/default.nix:21]:
```suggestion
  subPackages = [ "cmd/rain" ];
```

File: pkgs/development/tools/rain/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    command = "rain --version";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    description = "A development workflow tool for working with AWS CloudFormation.";
+    homepage = "https://github.com/aws-cloudformation/rain";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ jiegec ];
+    platforms = lib.platforms.unix;
```
Comment (azahi) [inline pkgs/development/tools/rain/default.nix:36]:
```suggestion
    platforms = platforms.unix;
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rain/default.nix (hunk 1 above)
Comment (jiegec) [reply pkgs/development/tools/rain/default.nix]: Fixed.

File: pkgs/development/tools/rain/default.nix (hunk 2 above)
Comment (jiegec) [reply pkgs/development/tools/rain/default.nix]: Fixed.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 182351` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rain</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

File: pkgs/development/tools/rain/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  ldflags = [ "-s" "-w" ];
+
+  passthru.tests.version = testers.testVersion {
+    package = rain;
+    command = "rain --version";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    description = "A development workflow tool for working with AWS CloudFormation.";
```
Comment (smancill) [inline pkgs/development/tools/rain/default.nix:32]:
```suggestion
    description = "A development workflow tool for working with AWS CloudFormation";
```

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 182351` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rain</li>
  </ul>
</details>

File: pkgs/development/tools/rain/default.nix (hunk 3 above)
Comment (jiegec) [reply pkgs/development/tools/rain/default.nix]: Fixed.

**Review by smancill — APPROVED** [review]

**Pull request merged**